tools/wpt/utils.py

```diff
@@ -55,9 +56,39 @@ def unzip(fileobj, dest=None, limit=None):
         for info in zip_data.infolist():
             if limit is not None and info.filename not in limit:
                 continue
-            zip_data.extract(info, path=dest)
-            perm = info.external_attr >> 16 & 0x1FF
-            os.chmod(os.path.join(dest, info.filename), perm)
+            # external_attr has a size of 4 bytes and the info it contains depends on the system where the ZIP file was created.
+            # - If the Zipfile was created on an UNIX environment, then the 2 highest bytes represent UNIX permissions and file
+            #   type bits (sys/stat.h st_mode entry on struct stat) and the lowest byte represents DOS FAT compatibility attributes
+            #   (used mainly to store the directory bit).
+            # - If the ZipFile was created on a WIN/DOS environment then the lowest byte represents DOS FAT file attributes
+            #   (those attributes are: directory bit, hidden bit, read-only bit, system-file bit, etc).
+            # More info at https://unix.stackexchange.com/a/14727 and https://forensicswiki.xyz/page/ZIP
+            # So, we can ignore the DOS FAT attributes because python ZipFile.extract() already takes care of creating the directories
+            # as needed (both on win and *nix) and the other DOS FAT attributes (hidden/read-only/system-file/etc) are not interesting
+            # here (not even on Windows, since we don't care about setting those extra attributes for our use case).
+            # So we do this:
+            #   1. When uncompressing on a Windows system we just call to extract().
+            #   2. When uncompressing on an Unix-like system we only take care of the attributes if the zip file was created on an
+            #      Unix-like system, otherwise we don't have any info about the file permissions other than the DOS FAT attributes,
+            #      which are useless here, so just call to extract() without setting any specific file permission in that case.
+            if info.create_system == 0 or sys.platform == 'win32':
```

Comment:
I guess this means we don't even try to create symlinks on Windows, which aiui is increasingly possible these days. But since we're mainly interested in the symlink support for Linux CI I guess that's not an immediate concern.

Reply:
An update: I have tested how symlinks and ZIP files work on Windows. I tested with the Windows internal ZIP tool as well as with WinZIP, WinRAR and 7-zip

I found this:

* Uncompressing a ZIP file created on Linux containing symlinks:

  * Windows Internal ZIP, WinZIP and WinRAR: For the symlinks it simply creates text files containing the link destination inside the text file
  * 7-zip: It creates empty files for the symlinks if the user doesn't have the symlink permission (by default users on Windows don't have this permission) and if the user has the permission then it properly creates the symlinks

* Compressing a directory that has symlinks on Windows into a ZIP:
  * All of the 4 tools tested store a copy of the file instead of storing the symlink.
     * So it seems storing symlinks inside a ZIP file is still not implemented on Windows.

So the current code is behaving like Windows Internal ZIP, WinZIP and WinRAR. This is good enough to me.